### PR TITLE
io.kinvolk.headlamp

### PR DESCRIPTION
**Pull request by joaquimrocha**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). `*`
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

`*` The company I work for owns the domain and I am allowed to file this submission. Not sure whether I should check the box (maybe the sentence needs to be updated to account for having the permission to use the domain).

Assuming I will have the permissions in the repo for inviting members, I will then add the Headlamp developers who need to have write permissions.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build io.kinvolk.headlamp

**Comment by barthalion** [comment]
bot, build io.kinvolk.Headlamp

File: io.kinvolk.Headlamp.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  <url type="homepage">https://github.com/kinvolk/headlamp/</url>
+  <url type="bugtracker">https://github.com/kinvolk/headlamp/issues</url>
+  <url type="help">https://github.com/kinvolk/headlamp#get-involved</url>
+  <launchable type="desktop-id">io.kinvolk.Headlamp.desktop</launchable>
+
+  <releases>
+    <release version="0.3.0" date="2021-04-17" />
+    <release version="0.2.1" date="2021-02-03" />
+    <release version="0.2.0" date="2021-01-14" />
+    <release version="0.1.3" date="2021-11-27" />
```
Comment (hfiguiere) [inline io.kinvolk.Headlamp.appdata.xml:34]: This date is in the future, hence the build failure.

Comment (joaquimrocha) [reply io.kinvolk.Headlamp.appdata.xml:34]: Woops! Definitely we didn't do it in 7 months.

**Comment by joaquimrocha** [comment]
bot, build io.kinvolk.Headlamp

**Comment by joaquimrocha** [comment]
Arrgh, forgot to account for the linux-unpacked name on arm64.
I'll check it out later.

**Comment by barthalion** [comment]
> Arrgh, forgot to account for the linux-unpacked name on arm64.

It's fine to disable aarch64 for now, if you wish: https://github.com/flathub/flathub/wiki/App-Maintenance#dont-build-on-arm

**Comment by joaquimrocha** [comment]
I thought I could just make the changes to include it in the flatpak build, but it needs a patch and testing, so I will skip it for now.

**Comment by joaquimrocha** [comment]
bot, build io.kinvolk.Headlamp

**Comment by joaquimrocha** [comment]
bot, build io.kinvolk.Headlamp

**Comment by joaquimrocha** [comment]
The URL for testing the flatpak gives me 404...
Is there a way to a working ref?

**Comment by barthalion** [comment]
Sorry, we were having some technical issues today. Let's try again.

bot, build io.kinvolk.Headlamp

**Comment by joaquimrocha** [comment]
Okay. Tried locally and I think I may have messed up something in a recent change. Give me a few minutes.

**Comment by joaquimrocha** [comment]
Yup. Had some unwanted changes when I committed the latest patch.

bot, build io.kinvolk.Headlamp

**Comment by barthalion** [comment]
You likely want to add `--device=dri` too. The build above launches, but is stuck fetching clusters. This is what I've found in terminal:

```
17:10:58.175 › server process stderr: /bin/sh: line 1: /app/headlamp-app/resources/server: No such file or directory
```

**Comment by joaquimrocha** [comment]
Yeah, it's not the dri AFAIU, but the server is not getting copied. I could reproduce it locally now. I'll follow up.

**Comment by joaquimrocha** [comment]
I had to use master instead of v0.3.0 because it has a couple of patches for improving the app build.
I could apply the patches but I think it's fine to use the master branch now and set it to a release tag again once we have a release (unless this is not recommended).

bot, build io.kinvolk.Headlamp

File: io.kinvolk.Headlamp.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      npm_config_no_save: 'true'
+      npm_config_loglevel: notice
+      GOPATH: /run/build/headlamp/
+      GO111MODULE: "auto"
+      GOFLAGS: "-mod=vendor"
+      DEBUG: electron-builder
+  sources:
+  - type: git
+    url: https://github.com/kinvolk/headlamp.git
+    branch: master
```
Comment (barthalion) [inline io.kinvolk.Headlamp.yaml:46]: Could you pin a specific commit here? Just so the build is reproducible. Other than that, LGTM.

Comment (joaquimrocha) [reply io.kinvolk.Headlamp.yaml:46]: Makes sense. Will do. Thanks.

**Comment by joaquimrocha** [comment]
bot, build io.kinvolk.Headlamp

**Comment by joaquimrocha** [comment]
Should be all fine now.

**Pull request closed without merging**